packages/squiggle-lang/src/rescript/Reducer/Reducer_MathJs/Reducer_MathJs_Parse.res

```diff
@@ -0,0 +1,158 @@
+/*
+  MathJs Nodes
+  We make MathJs Nodes strong-typed
+*/
+module AE = Reducer_Extra_Array
+module JsG = Reducer_Js_Gate
+module Rerr = Reducer_Error
+
+type reducerError = Rerr.reducerError
+
+type node = {
+  "type": string,
+  "isNode": bool,
+  "comment": string
+}
+type arrayNode = {
+  ...node,
+  "items": array<node>
+}
+//assignmentNode
+//blockNode
+//conditionalNode
+type constantNode = {
+  ...node,
+  "value": unit
+}
+//functionAssignmentNode
+type functionNode = {
+  ...node,
+  "fn": string,
+  "args": array<node>
+}
+type indexNode = {
+  ...node,
+  "dimensions": array<node>
+}
+type objectNode = {
+  ...node,
+  "properties": Js.Dict.t<node>
+}
+type accessorNode = {
+  ...node,
+  "object": node,
+  "index": indexNode
+}
+type operatorNode = {
+  ...functionNode,
+  "op": string,
+}
+
+//parenthesisNode
+type parenthesisNode = {
+  ...node,
+  "content": node
+}
+//rangeNode
+//relationalNode
+type symbolNode = {
+  ...node,
+  "name": string
+}
+
+external castAccessorNode: node => accessorNode = "%identity"
+external castArrayNode: node => arrayNode = "%identity"
+external castConstantNode: node => constantNode = "%identity"
+external castFunctionNode: node => functionNode = "%identity"
+external castIndexNode: node => indexNode = "%identity"
+external castObjectNode: node => objectNode = "%identity"
+external castOperatorNode: node => operatorNode = "%identity"
+external castOperatorNodeToFunctionNode: operatorNode => functionNode = "%identity"
+external castParenthesisNode: node => parenthesisNode = "%identity"
+external castSymbolNode: node => symbolNode = "%identity"
+
+/*
+  MathJs Parser
+*/
+@module("mathjs") external parse__: string => node = "parse"
+
+let parse = (expr: string): result<node, reducerError> =>
+  try {
+    Ok(parse__(expr))
+  } catch {
+  | Js.Exn.Error(obj) =>
+    RerrJs(Js.Exn.message(obj), Js.Exn.name(obj))->Error
+  }
+
+type mjNode =
```

Comment:
This is another example of a compressed acronym. I'd replace with ``mathjsNode``

Reply:
RESOLVED